.github/workflows/go.yml

```diff
@@ -1,25 +1,129 @@
-name: Go
+name: Build, Test and Release
 
 on:
   push:
-    branches: [ main ]
+    branches:
+      - main
+    tags:
+      - v*.*.*
   pull_request:
-    branches: [ main ]
+    branches:
+      - main
 
 jobs:
 
   build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+
+    env:
+      GO_LDFLAGS: "-w -s"
+      GO11MODULE: "on"
+      GINKGO_FLAGS: "--randomizeAllSpecs --randomizeSuites --failOnPending --trace --race --progress --compilers=2 --nodes=2"
+      VERSION: ${{ startsWith(github.ref, 'refs/tags/v') && github.ref_name || format('gh-action-{0}', github.sha) }}
+
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # renovate: tag=v2
+
+      - name: Setup Cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg
+          key: ${{ runner.os }}-build-${{ runner.arch }}
+
+      - name: Set up Go
+        uses: actions/setup-go@331ce1d993939866bb63c32c6cbbfd48fa76fc57 # renovate: tag=v2
+        with:
+          go-version: 1.17
+
+      - name: Build Binary
+        run: |
+          make generate
+          if ! git diff --exit-code --name-only; then
+            git diff
+            echo 'Run "make generate" and add generated changes to this PR'
+            exit 1
+          fi
+          # Build kubecfg
+          make all
+
+      - name: Info Binary
+        run: |
+          #ldd ./kubecfg || otool -L ./kubecfg # What's the purpose of this ? it currently fail since we don't produce a dynamic executable
+          ./kubecfg help
+          ./kubecfg version
+
+      - name: Test Binary
+        run: |
+          make test
+          make vet
+
+      # Only run integration on linux matrix job
+      - name: Create k8s Kind Cluster
+        uses:  engineerd/setup-kind@v0.5.0
+        if: matrix.os == 'ubuntu-latest'
+        with:
+          version: "v0.11.1"
```

Comment:
whose version is this? kind's?

Reply:
yes, the image variable below define the k8s version